pkg/etcdenvvar/etcd_env.go

```diff
@@ -243,6 +244,25 @@ func getHardwareSpeedValues(envVarContext envVarContext) (map[string]string, err
 	return hwspeedhelpers.HardwareSpeedToEnvMap(speed)
 }
 
+func getEtcdDBSize(envVarContext envVarContext) (map[string]string, error) {
+	etcd, err := envVarContext.etcdLister.Get("cluster")
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve etcd CR: %v", err)
+	}
+
+	backendQuotaGiB := int64(8)
+	enabled, err := backendquotahelpers.IsBackendQuotaFeatureGateEnabled(envVarContext.featureGateAccessor)
```

Comment:
looks good, I would just error out before setting any wrong value. 
```suggestion
	enabled, err := backendquotahelpers.IsBackendQuotaFeatureGateEnabled(envVarContext.featureGateAccessor)
	if err != nil {
		return nil, fmt.Errorf("failed check backend quota enabled: %w", err)
	}
```

Reply:
but if we error, then we return `nil` envVarMap 

is this expected behaviour ?

Reply:
I think so, it would not create a new revision:
https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/etcdenvvar/envvarcontroller.go#L163-L165

Reply:
done 👍🏽